packages/shared/src/hooks/bookmark/useCreateBookmarkFolder.ts

```diff
@@ -0,0 +1,58 @@
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+import { useCallback } from 'react';
+import { BookmarkFolder, createBookmarkFolder } from '../../graphql/bookmarks';
+import { useLogContext } from '../../contexts/LogContext';
+import { LogEvent } from '../../lib/log';
+import { useToastNotification } from '../useToastNotification';
+import { generateQueryKey, RequestKey } from '../../lib/query';
+
+type CreateBookmarkFolderProps = Pick<BookmarkFolder, 'name' | 'icon'>;
+
+interface UseCreateBookmarkFolder {
+  isPending: boolean;
+  createFolder: (
+    folder: CreateBookmarkFolderProps,
+  ) => Promise<BookmarkFolder | null>;
+}
+
+export const useCreateBookmarkFolder = (): UseCreateBookmarkFolder => {
+  const { logEvent } = useLogContext();
+  const { displayToast } = useToastNotification();
+  const queryClient = useQueryClient();
+
+  const { mutateAsync, isPending } = useMutation({
+    mutationFn: createBookmarkFolder,
+  });
+
+  const createFolder: UseCreateBookmarkFolder['createFolder'] = useCallback(
+    async (folder) => {
+      try {
```

Comment:
Does the `useMutation` don't have the `onError` and `onSuccess`? I think we can move the parts from here to those props.